flink-core/src/main/java/org/apache/flink/configuration/HistoryServerOptions.java

```diff
@@ -89,6 +89,16 @@
 			.withDescription("Enable HTTPs access to the HistoryServer web frontend. This is applicable only when the" +
 				" global SSL flag security.ssl.enabled is set to true.");
 
+	/**
+	 * Specifies whether should HistoryServer process job archive deletion. If this option is enabled
+	 * then deleted job archives are also deleted from HistoryServer. By default is disabled.
+	 */
+	public static final ConfigOption<Boolean> HISTORY_SERVER_PROCESS_JOB_DELETION =
+		key("historyserver.web.process-job-deletion")
+			.defaultValue(false)
+			.withDescription("Specifies whether should HistoryServer process job archive deletion." +
```

Comment:
Whether HistoryServer should cleanup jobs, that are no longer present in `historyserver.archive.fs.dir`.

Reply:
Changed, thank you